src/marqo/instance_mappings.py

```diff
@@ -0,0 +1,33 @@
+from abc import ABC, abstractmethod
+
+from typing import Dict, Optional
+
+
+class InstanceMappings(ABC):
+    """
+    Abstract class for mapping index names to URLs.
+
+    Implementations of this class must ensure at least asymptotically constant computational complexity for all methods.
+    An inefficient implementation of this class can cause the Marqo client to be slow.
```

Comment:
@farshidz we had an idea for a clarifying sentence after this first sentence ("... as the number of requests per second grow, methods of this implementations of this class should tend towards constant time, given typical usage patterns for a target usecase."). 

Good if @farshidz could fill in details/confirm.